main/src/library/flix/actor/ActorSystem.flix

```diff
@@ -0,0 +1,183 @@
+///
+/// An actor system is a type alias for a record that consists of four fields:
+///
+/// @field `actors`          that declares the actors in the system.
+/// @field `startPolicy`     that describes the policy for how actors are started.
+/// @field `restartPolicy`   that describes the policy for how actors are restarted if they crash.
+/// @field `shutdownPolicy`  that describes the policy for how actors are stopped.
+/// @field `control`         the runtime control channel of this actor system
+/// @field `actors`          the actors that are running
+///
+type alias ActorSystem = {
+    actors: ActorPolicy,
+    startPolicy: ActorPolicy,
+    restartPolicy: ActorPolicy,
+    shutdownPolicy: ActorPolicy
+}
+
+///
+/// A running actor system is a type alias for a channel of actor events.
+///
+type alias RunningActorSystem = Channel[SystemEvent]
+
+///
+/// An actor is identified by its mailbox
+///
+type alias ActorRef = Mailbox // TODO: Quentin: Maybe a name and a mailbox?
+
+namespace ActorSystem {
+
+    ///
+    /// Returns an actor system with a default configuration.
+    ///
+    pub def default(): ActorSystem = {
+        actors          = #{},
+        startPolicy     = defaultStartPolicy(),
+        shutdownPolicy  = defaultShutdownPolicy(),
+        restartPolicy   = defaultRestartPolicy()
+    }
+
+    ///
+    /// Returns the default mailbox capacity.
+    ///
+    pub def defaultMailboxCapacity(): Int = 1000 * 1000
+
+    enum InternalActorState {
+      ///
+      /// An actor that has not been started previously, and has no mailbox
+      ///
+      case NotStarted,
+      ///
+      /// An actor that has been started and been assigned a mailbox. The actor could be stopped or crashed now.
+      ///
+      case Started(State, ActorRef)
+    }
+
+    def statesOfActors(actors: Map[Str, (InternalActorState, Actor)]): ActorPolicy =
+      Map.foldLeftWithKey((actorName, states, actor) ->
+        match fst(actor) {
+          case NotStarted => states <+> #{ActorState(actorName, Stopped).}
+          case Started(st, _) => states <+> #{ActorState(actorName, st).}
+        }, #{}, actors)
+
+
+  def actorSystem(s: ActorSystem, actors: Map[Str, (InternalActorState, Actor)], control: Channel[SystemEvent]): Unit = {
+      match <- control {
+        case SystemStart =>
+          let currentState = statesOfActors(actors);
+          let actors2 = fold Start actors ((actorName, m) -> match Map.get(actorName, actors) {
```

Comment:
I tried to write out some ideas in ActorPolicy for how this "table" could look, i.e. what should happen when you try to say stop an already stopped actor.